2018/f8801_2019.py

```diff
@@ -0,0 +1,90 @@
+from form import Form, FilingStatus
+
+class F8801_2019(Form):
+    """Form 8801, Credit for Prior Year Minimum Tax
+
+    This form is for 2019, but since it depends on values
+    from 2018, it is generated along with 2018's
+    forms. Line 21 from this form must be manually entered into
+    2019's input dictionary.
+    """
+    def __init__(f, inputs, f1040, f6251, f8801, sched_d):
+        super(F8801_2019, f).__init__(inputs)
+        f['1'] = f6251.rowsum(['1', '2e'])
+        f['2'] = f6251.rowsum(['2a', '2b', '2c', '2d', '2g', '2h'])
+        # TODO: Line 3, minimum tax credit net operating loss deduction
+        f['4'] = max(0, f['1'] + f['2'] + f['3'])
+        assert(inputs['status'] != FilingStatus.SEPARATE or f['4'] <= 718800)
+        f.comment['15'] = 'Net minimum tax on exclusion items'
+        if f['4'] == 0:
+            f['15'] = 0
+        else:
+            f['5'] = f6251.EXEMPTIONS[inputs['status']]
+            f['6'] = f6251.EXEMPT_LIMITS[inputs['status']]
+            f['7'] = max(0, f['4'] - f['6'])
+            f['8'] = f['7'] * 0.25
+            f['9'] = max(0, f['5'] - f['8'])
+            f['10'] = max(0, f['4'] - f['9'])
+            if f['10'] == 0:
+                f['15'] = 0
+            else:
+                # TODO: form 2555
+                if (f1040['13'] and not sched_d.mustFile()) or f1040['9b'] or \
```

Comment:
'13' should be 's13' and '9b' should be '3a'

Reply:
good catch, done.